crates/holochain/src/conductor/conductor/tests/agent_key_revocation.rs

```diff
@@ -0,0 +1,199 @@
+use holo_hash::{ActionHash, AgentPubKey};
+use holochain_conductor_services::{DpkiServiceError, KeyState};
+use holochain_types::app::{AppError, CreateCloneCellPayload};
+use holochain_wasm_test_utils::TestWasm;
+use holochain_zome_types::action::ActionType;
+use holochain_zome_types::dependencies::holochain_integrity_types::DnaModifiersOpt;
+use holochain_zome_types::record::Record;
+use holochain_zome_types::timestamp::Timestamp;
+use matches::assert_matches;
+use rusqlite::Row;
+
+use crate::conductor::api::error::ConductorApiError;
+use crate::conductor::{conductor::ConductorError, CellError};
+use crate::core::workflow::WorkflowError;
+use crate::core::{SysValidationError, ValidationOutcome};
+use crate::sweettest::{
+    SweetConductor, SweetConductorConfig, SweetDnaFile, SweetInlineZomes, SweetZome,
+};
+
+#[tokio::test(flavor = "multi_thread")]
+async fn revoke_agent_key() {
+    let mut conductor = SweetConductor::from_standard_config().await;
+    let (dna_file_1, _, coordinator_zomes_1) =
+        SweetDnaFile::unique_from_test_wasms(vec![TestWasm::Create]).await;
+    let (dna_file_2, _, coordinator_zomes_2) =
+        SweetDnaFile::unique_from_test_wasms(vec![TestWasm::Create]).await;
+    let role_1 = "role_1";
+    let role_2 = "role_2";
+    let app = conductor
+        .setup_app(
+            "",
+            [
+                &(role_1.to_string(), dna_file_1.clone()),
+                &(role_2.to_string(), dna_file_2.clone()),
+            ],
+        )
+        .await
+        .unwrap();
+    let agent_key = app.agent().clone();
+    let zome_1 = SweetZome::new(
+        app.cells()[0].cell_id().clone(),
+        coordinator_zomes_1[0].name.clone(),
+    );
+    let zome_2 = SweetZome::new(
+        app.cells()[1].cell_id().clone(),
+        coordinator_zomes_2[0].name.clone(),
+    );
+    let create_fn_name = "create_entry";
+    let read_fn_name = "get_post";
+
+    // no agent key provided, so DPKI should be installed
+    // and the generated agent key be valid
```

Comment:
Is dpki installed as a consequence of no agent key being provided? That's how this is reading to me

```suggestion
    // no agent key provided, so the running DPKI service should be used to generate a valid agent key
```

Reply:
no, it's not. It took me a while to understand how the DPKI service works and this comment was from early on.